mindstream-session.el

```diff
@@ -85,10 +85,17 @@ This only removes implicit versioning. It does not close any open
 buffers at the SESSION path."
   (interactive (list (completing-read "Which session? "
                                       mindstream-active-sessions
-                                      (lambda (session)
-                                        (not
-                                         (string-prefix-p mindstream-path
-                                                          session))))))
+                                      ;; This excludes all anonymous
+                                      ;; sessions, which we should be
+                                      ;; able to also end.
+
+                                      ;; (lambda (session)
+                                      ;;   (not
+                                      ;;    (string-prefix-p mindstream-path
```

Comment:
There's typically no need to end an anonymous session (just kill the buffer and forget about it), so that's why I didn't include this "noise" in the selection candidates.

Yet, it's true that currently, killing buffers does not make the session inactive (from the perspective of `mindstream-active-sessions` which determines versioning), so the only way to end them is to save the session to a non-anonymous one. We could allow users to manually end such sessions, but that wouldn't be ideal.

One option (just mentioning, not for this PR) is to add a `kill-buffer-hook` that will implicitly end the session if there are no buffers in it remaining open.

Another thought is, [I am suspicious of `mindstream-active-sessions`](https://github.com/countvajhula/mindstream/issues/18) to begin with, as having this kind of global state could be bug-prone. I'm thinking that if we make sessions live in dedicated Git branches (so that `mindstream-begin-session` will call `git checkout -b mindstream-<unique-id+timestamp>`. Then, whether a session is "active" could be determined by just checking whether the current Git branch begins with "mindstream", so that there is no need for global state in mindstream, and then I think we wouldn't need the `kill-buffer` hook either. I'll add this as a comment on the relevant issue to keep things in context.